### DISCUSSION TITLE
Internal_temp or external_temp are not set

### DISCUSSION THREAD

**Discussion by pbranly** [jmcollin78/versatile_thermostat#173] [Q&A] [answered]
Hi
I have configured versatile thermostat with 3 TRVs (thermostatc valves)
I works but I get this error in the HA logs:
Logger: custom_components.versatile_thermostat.pi_algorithm
Source: custom_components/versatile_thermostat/pi_algorithm.py:42
Integration: Versatile Thermostat configuration ([documentation](https://github.com/jmcollin78/versatile_thermostat), [issues](https://github.com/jmcollin78/versatile_thermostat/issues))
First occurred: 15:20:35 (12 occurrences)
Last logged: 15:21:39
Internal_temp or external_temp are not set. Regulation will be suspended

What is the reason for ?
Thanks
Phil

**Comment by jmcollin78** [comment, marked as answer]
This is because as said in the error message, the internal temperature or the external temperature are not set.
Check your temperature sensor configuration and that they are working (giving a temperature).

It is possible at startup to have this message while waiting for temperature sensors but this should not persist

**Reply by pbranly** [reply]
Yes it seems and happen at the beginning only

First occurred: 15:20:35 (12 occurrences)
Last logged: 15:21:39

So 12 occurrences during the first minute then nothing up to 16:52 

Thanks 
Phil

**Comment by maia** [comment]
@jmcollin78, maybe the following change in `pi_algorithm.py` could help interpreting the issue. It's very common with Zigbee/Thread networks for sensors that are far away and have zigbee/thread routers inbetween, because they need to boot first.

```py
if internal_temp is None:
    _LOGGER.warning("Temporarily skipping the self-regulation algorithm while the configured sensor for room temperature is unavailable")
    return self.target_temp
if external_temp is None:
    _LOGGER.warning("Temporarily skipping the self-regulation algorithm while the configured sensor for outdoor temperature is unavailable")
    return self.target_temp
```

**Reply by jmcollin78** [reply]
yep. will take this for next release.
